src/hotspot/cpu/s390/macroAssembler_s390.cpp

```diff
@@ -5803,3 +5803,38 @@ void MacroAssembler::lightweight_unlock(Register obj, Register hdr, Register tmp
   z_alsi(in_bytes(JavaThread::lock_stack_top_offset()), Z_thread, -oopSize);  // pop object
   z_cr(tmp, tmp); // set CC to EQ
 }
+
+void MacroAssembler::population_count(Register r_dst, Register r_src, Register r_tmp, bool is_long) {
+  BLOCK_COMMENT("population_count {");
+
+  if (VM_Version::has_MiscInstrExt3()) {
+    z_popcnt(r_dst, r_src, 8);
+  } else {
+
+    z_illtrap(48); // fixme: remove
```

Comment:
I'll remove this before making this PR ready for review. As of now I'm using z16 machine, So this is not producing any error.